optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/result/BenchmarkResultIO.java

```diff
@@ -78,8 +78,7 @@ public void writePlannerBenchmarkResult(File benchmarkReportDirectory,
             throw new IllegalArgumentException("The benchmarkDirectory (" + benchmarkDirectory
                     + ") is not a directory.");
         }
-        File[] benchmarkReportDirectories = benchmarkDirectory.listFiles((FileFilter) DirectoryFileFilter.INSTANCE);
-        Arrays.sort(benchmarkReportDirectories);
+        File[] benchmarkReportDirectories = retrieveSortedBenchmarkReportDirectories(benchmarkDirectory);
```

Comment:
3 lines above there is a "benchmarkDirectory.exists()" check, so this will never return null (unless in a race condition at which time it should fail fast and NPE is ok enough for me).

Can we revert this change and tell Findbugs to exclude it?

Reply:
Oh, isn't this the same situation as the other one in this PR that I already commented on?

If I understand it correctly, the null risk is not with `benchmarkDirectory` but with the result of `listFiles()` which is not guaranteed to be non-null. Replacing it with an empty array looks like an elegant solution that allows the code below to work with it safely without being enclosed with a null-checking `if` statement.

Reply:
@yurloc is correct - it's about the listFiles() result and listFiles() returns null instead of throwing IOException, so we need the null check even if we check for directory. I will leave this null check in place.

Reply:
@rsynek Let's jump on bluejeans to discuss this one